### PR TITLE
WIP Na/wire up bet manager card

### PR DESCRIPTION
**Pull request by devnic113**
- sync USD, ETH and WEI
- wire up Place Bet
- Known issues: sometimes the Place Bet button does not work/submit a bet. I noticed this happening after choosing a new Hash character (binary or hex)

### CODE REVIEW

File: packages/nextjs/components/BetManager.tsx [hunk 1]
```diff
@@ -12,13 +95,28 @@ function setBetManagerActiveTab(activeTab: BetManagerTabEnum) {
...
+    return parseInt(s, 16);
+  }
+  return parseInt(s.join(""), 2);
+}
+
 const BetManager = ({ writePlaceBet }: { writePlaceBet: any }) => {
   const [activeTab, setActiveTab] = useState<BetManagerTabEnum>(() => {
     return (localStorage.getItem("betManagerActiveTab") as BetManagerTabEnum) || BetManagerTabEnum.PLACE_BET;
   });
+  const ethPrice = 3800;
```
Comment (devnic113) [inline packages/nextjs/components/BetManager.tsx:116]: hard coded for now but we should consume a ws stream

Comment (devnic113) [reply packages/nextjs/components/BetManager.tsx:116]: and if we do that, we can wire it up to the setEthPrice hook in the betAMount reducer

File: packages/nextjs/components/BetManager.tsx [hunk 2]
```diff
@@ -1,8 +1,91 @@
...
+
+  return {
+    betAmount,
+    updateUsd: (usd: number) => dispatch({ type: "SET_USD", payload: usd }),
+    updateEth: (eth: number) => dispatch({ type: "SET_ETH", payload: eth }),
+    updateWei: (wei: number) => dispatch({ type: "SET_WEI", payload: wei }),
+    updateEthPrice: (ethPrice: number) => dispatch({ type: "SET_ETH_PRICE", payload: ethPrice }),
+  };
+};
+
```
Comment (wmbryce) [inline packages/nextjs/components/BetManager.tsx:88]: This is pretty cool. I haven't used this hook before I'm pretty sure its new in React 18 or 19. Its pretty sick.

File: packages/nextjs/components/BetManager.tsx (hunk 1 above)
Comment (wmbryce) [reply packages/nextjs/components/BetManager.tsx:116]: Check out: useInitializeNativeCurrencyPrice
It looks like this fetches the current price from Uniswap.

**Comment by wmbryce** [comment]
The UI doesn't throw an error when you have a bet amount of $0

**Pull request merged**